### PR TITLE
[reconfigurator] Enable clickhouse services via clickhouse-admin

### PR DESCRIPTION
**Pull request by karencfv**
## Overview

This commit makes a few changes to the way the `clickhouse_server` and `clickhouse_keeper` SMF services are launched:

- They are now disabled by default on zone boot.
- They are enabled by clickhouse-admin after the configuration files have been generated.

Related: https://github.com/oxidecomputer/omicron/issues/5999

### CODE REVIEW

File: clickhouse-admin/src/http_entrypoints.rs [hunk 1]
```diff
@@ -31,10 +32,18 @@ impl ClickhouseAdminApi for ClickhouseAdminImpl {
...
         let output =
             ctx.clickward().generate_server_config(replica_server.settings)?;
+
+        // Once we have generated the client we can safely enable the clickhouse_server service
+        let fmri = "svc:/oxide/clickhouse_server:default".to_string();
+        Svcadm::enable_service(fmri).map_err(|err| HttpError {
+            status_code: http::StatusCode::INTERNAL_SERVER_ERROR,
+            error_code: Some(String::from("Internal")),
+            external_message: format!("{err}"),
+            internal_message: format!("{err}"),
```
Comment (karencfv) [inline clickhouse-admin/src/http_entrypoints.rs:44]: I don't love this, should probably find a way to make it nicer

**Comment by karencfv** [comment]
@andrewjstone, please feel free to directly modify, merge, or close this PR directly, as I'll be out for two weeks :)

File: clickhouse-admin/src/http_entrypoints.rs [hunk 2]
```diff
@@ -25,28 +26,34 @@ enum ClickhouseAdminImpl {}
...
     }
 
-    async fn generate_keeper_config(
+    async fn generate_keeper_config_and_enable(
         rqctx: RequestContext<Self::Context>,
         body: TypedBody<KeeperConfigurableSettings>,
     ) -> Result<HttpResponseCreated<KeeperConfig>, HttpError> {
         let ctx = rqctx.context();
         let keeper = body.into_inner();
-        // TODO(https://github.com/oxidecomputer/omicron/issues/5999): Do something
```
Comment (andrewjstone) [inline clickhouse-admin/src/http_entrypoints.rs:47]: We still want to use the generation number to prevent unnecessary config generations on every push.

**Review by andrewjstone — APPROVED** [review]
Looks good! I'm going to merge it in and build upon it. Last thing needed for a basic demo is some OMDB support!

**Pull request merged**